monai/apps/reconstruction/mri_array.py

```diff
@@ -0,0 +1,282 @@
+                The last spatial dim is selected for sampling. For the fastMRI
+                dataset, k-space has the form (...,num_slices,num_coils,H,W)
+                and sampling is done along W. For a general 3D data with the
+                shape (...,num_coils,H,W,D), sampling is done along D.
+            is_complex: if True, then the last dimension will be reserved for
+                real/imaginary parts.
+        """
+        if len(center_fractions) != len(accelerations):
+            raise ValueError(
+                "Number of center fractions \
+                should match number of accelerations"
+            )
+
+        self.center_fractions = center_fractions
+        self.accelerations = accelerations
+        self.spatial_dims = spatial_dims
+        self.is_complex = is_complex
+
+    @abstractmethod
+    def __call__(self, kspace: NdarrayOrTensor):
+        """
+        This is an extra instance to allow for defining new mask generators.
+        For creating other mask transforms, define a new class and simply
+        override __call__. See an example of this in 'RandomMaskFunc.'
+
+        Args:
+            kspace: The input k-space data. The shape is (...,num_coils,H,W,2)
+                for complex 2D inputs and (...,num_coils,H,W,D) for real 3D
+                data.
+        """
+        raise NotImplementedError
+
+    def randomize_choose_acceleration(self) -> Sequence[float]:
+        """
+        If multiple values are provided for center_fractions and
+        accelerations, this function selects one value uniformly
+        for each training/test sample.
+
+        Returns:
+            A tuple containing
+                (1) center_fraction: chosen fraction of center kspace
+                lines to exclude from under-sampling
+                (2) acceleration: chosen acceleration factor
+        """
+        choice = self.R.randint(0, len(self.accelerations))
+        center_fraction = self.center_fractions[choice]
+        acceleration = self.accelerations[choice]
+        return center_fraction, acceleration
+
+
+class RandomKspaceMask(KspaceMask, Transform):
+    """
+    This k-space mask transform under-samples the k-space according to a
+    random sampling pattern. Precisely, it uniformly selects a subset of
+    columns from the input k-space data. If the k-space data has N columns,
+    the mask picks out:
+    1. N_low_freqs = (N * center_fraction) columns in the center
+    corresponding to low-frequencies
+    2. The other columns are selected uniformly at random with a probability
+    equal to:
+    prob = (N / acceleration - N_low_freqs) / (N - N_low_freqs).
+    This ensures that the expected number of columns selected is equal to
+    (N / acceleration)
+
+    It is possible to use multiple center_fractions and accelerations,
+    in which case one possible (center_fraction, acceleration) is chosen
+    uniformly at random each time the transform is called.
+
+    Example:
```

Comment:
Have you tested `"make html"`? I was wondering if the current format can produce well-formatted docs...